subsys/bluetooth/host/audio/iso.c

```diff
@@ -0,0 +1,1075 @@
+
+	err = bt_hci_cmd_send_sync(BT_HCI_OP_LE_SET_CIG_PARAMS, buf, &rsp);
+	if (err) {
+		return NULL;
+	}
+
+	return rsp;
+}
+
+int bt_conn_bind_iso(struct bt_iso_create_param *param)
+{
+	struct bt_conn *conn;
+	struct net_buf *rsp;
+	struct bt_hci_rp_le_set_cig_params *cig_rsp;
+	int i, err;
+
+	/* Check if controller is ISO capable */
+	if (!BT_FEAT_LE_CIS_MASTER(bt_dev.le.features)) {
+		return -ENOTSUP;
+	}
+
+	if (!param->num_conns || param->num_conns > CONFIG_BT_MAX_ISO_CONN) {
+		return -EINVAL;
+	}
+
+	/* Assign ISO connections to each LE connection */
+	for (i = 0; i < param->num_conns; i++) {
+		conn = param->conns[i];
+
+		if (conn->type != BT_CONN_TYPE_LE) {
+			return -EINVAL;
+		}
+
+		conn = bt_conn_add_iso(conn);
+		if (!conn) {
+			return -ENOMEM;
+		}
+
+		/* TODO: Figure out if ID was already assigned? */
+		conn->iso.cig_id = param->id;
+		conn->iso.cis_id = bt_conn_index(conn);
+
+		param->conns[i] = conn;
+	}
+
+	rsp = hci_le_set_cig_params(param);
+	if (!rsp) {
+		err = -EIO;
+		goto failed;
+	}
+
+	cig_rsp = (void *)rsp->data;
+
+	if (rsp->len < sizeof(cig_rsp) ||
+	    cig_rsp->num_handles != param->num_conns) {
+		BT_WARN("Unexpected response to hci_le_set_cig_params");
+		err = -EIO;
+	}
+
+	for (i = 0; i < cig_rsp->num_handles; i++) {
+		/* Assign the connection handle */
+		param->conns[i]->handle = cig_rsp->handle[i];
+	}
+
+	net_buf_unref(rsp);
+
+	return 0;
+
+failed:
+	for (i = 0; i < param->num_conns; i++) {
+		bt_iso_cleanup(param->conns[i]);
+	}
+
+	return err;
+}
+
+static int hci_le_create_cis(struct bt_conn **conn, uint8_t num_conns)
+{
+	struct bt_hci_cis *cis;
+	struct bt_hci_cp_le_create_cis *req;
+	struct net_buf *buf;
+	int i;
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_LE_CREATE_CIS,
+				sizeof(*req) + sizeof(*cis) * num_conns);
+	if (!buf) {
+		return -ENOBUFS;
+	}
+
+	req = net_buf_add(buf, sizeof(*req));
+
+	memset(req, 0, sizeof(*req));
+
+	req->num_cis = num_conns;
+
+	/* Program the cis parameters */
+	for (i = 0; i < num_conns; i++) {
+		cis = net_buf_add(buf, sizeof(*cis));
+
+		memset(cis, 0, sizeof(*cis));
+
+		cis->cis_handle = sys_cpu_to_le16(conn[i]->handle);
+		cis->acl_handle = sys_cpu_to_le16(conn[i]->iso.acl->handle);
+	}
+
+	return bt_hci_cmd_send_sync(BT_HCI_OP_LE_CREATE_CIS, buf, NULL);
+}
+
+int bt_conn_connect_iso(struct bt_conn **conns, uint8_t num_conns)
+{
+	int i, err;
+
+	/* Check if controller is ISO capable */
+	if (!BT_FEAT_LE_CIS_MASTER(bt_dev.le.features)) {
+		return -ENOTSUP;
+	}
+
+	if (num_conns > CONFIG_BT_MAX_ISO_CONN) {
+		return -EINVAL;
+	}
+
+	for (i = 0; i < num_conns; i++) {
+		if (conns[i]->type != BT_CONN_TYPE_ISO) {
+			return -EINVAL;
+		}
+	}
+
+	err = hci_le_create_cis(conns, num_conns);
+	if (err) {
+		return err;
+	}
+
+	/* Set connection state */
+	for (i = 0; i < num_conns; i++) {
+		bt_conn_set_state(conns[i], BT_CONN_CONNECT);
+	}
+
+	return 0;
+}
+
+static int hci_le_setup_iso_data_path(struct bt_conn *conn,
+				      struct bt_iso_data_path *path)
```

Comment:
It seems that the `bt_conn` argument is only used for the handle. Since BIS also use this command, I think the argument should be changed to a `uint16_t handle` rather than a `bt_conn` so that it applies for both CIS and BIS. 

Seems to also be existing for more functions in this file.

Reply:
I leaning towards making BIS map to a bt_conn as well since it is effectively an ISO connection we probably want to reuse the same code.